server/src/main/java/com/linecorp/centraldogma/server/internal/httpapi/ContentServiceV1.java

```diff
@@ -0,0 +1,334 @@
+                    throw new HttpResponseException(HttpStatus.BAD_REQUEST);
+                }
+                Exceptions.throwUnsafely(thrown);
+            }
+            return VOID;
+        });
+    }
+
+    /**
+     * PATCH /projects/{projectName}/repos/{repoName}/contents{path}
+     *
+     * <p>Patches a repository with the JSON_PATCH. Currently, only unremove repository operation is supported.
+     */
+    @ConsumeType("application/json-patch+json")
+    @Patch("regex:/projects/(?<projectName>[^/]+)/repos/(?<repoName>[^/]+)/contents(?<path>/.*$)")
+    public CompletionStage<EntryDto> patchFile(@Param("projectName") String projectName,
+                                                   @Param("repoName") String repoName,
+                                                   @Param("path") String path,
+                                                   @Param("revision") @Default("-1") String revision,
+                                                   AggregatedHttpMessage message) {
+        checkRepositoryExists(projectName, repoName);
+
+        final Change<?> change;
+        final CommitMessageDto commitMessage;
+        try {
+            final JsonNode jsonNode = Jackson.readTree(message.content().toStringUtf8());
+            checkArgument(jsonNode != null && jsonNode.get("patch") != null &&
+                          jsonNode.get("commitMessage") != null, "patch and commit message should be non-null");
+            change = getChangeOfPatch(path, jsonNode.get("patch"));
+            commitMessage = convertCommitMessage(jsonNode.get("commitMessage"));
+        } catch (IOException e) {
+            throw new HttpResponseException(HttpStatus.BAD_REQUEST);
+        }
+
+        final Repository repository = getRepository(projectName, repoName);
+        final long commitTimeMillis = System.currentTimeMillis();
+        return push(commitTimeMillis, AuthenticationUtil.currentAuthor(), repository, new Revision(revision),
+                    commitMessage, ImmutableList.of(change)).handle((resultRevision, thrown) -> {
+            if (thrown != null) {
+                if (thrown.getCause() instanceof ChangeConflictException) {
+                    throw new HttpResponseException(HttpStatus.BAD_REQUEST);
+                }
+                throw new HttpResponseException(HttpStatus.INTERNAL_SERVER_ERROR);
+            }
+            final EntryType type =
+                    change.type().contentType().isAssignableFrom(JsonNode.class) ? EntryType.JSON
+                                                                                 : EntryType.TEXT;
+            return DtoConverter.convert(resultRevision, path, type, commitTimeMillis);
+        });
+    }
+
+    private static Change<?> getChangeOfPatch(String path, JsonNode jsonNode) {
+        if (path.charAt(path.length() - 1) == '/') {
+            throw new HttpResponseException(HttpStatus.BAD_REQUEST);
```

Comment:
Perhaps we could rely on `Util.validateFilePath()` where necessary?

Reply:
Fixed. Thank you~!